lorry/hypervisor/hypevisor.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright (C) 2022-2023 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package hypervisor
+
+import (
+	"errors"
+
+	"github.com/dapr/kit/logger"
+	"github.com/spf13/pflag"
+)
+
+type Hypervisor struct {
+	Logger    logger.Logger
+	DBService *Daemon
+	Watcher   *Watcher
+}
+
+var hypervisor *Hypervisor
+
+func NewHypervisor(logger logger.Logger) (*Hypervisor, error) {
```

Comment:
Where  to process the os signals ?

Reply:
it's in the lorry main.go,  lorry will call hypervisor.StopAndWait() if it receives a TERM or INT signal.

Reply:
how many times do you permit for a process to restart and failed?

Reply:
there is no times limit for a process to restart and failed, if the process starts failed, hypervisor will sleep for a while and try to restart it util success.